src/api/cvc4cpp.h

```diff
@@ -590,6 +698,43 @@ class CVC4_PUBLIC Term
    */
   bool isNull() const;
 
+  /**
+   * @return true iff this Term has an operator
+   */
+  bool hasOperator() const;
+
+  /**
+   * @return true iff this Term has a builtin operator
+   *
+   * in this case, getKind() will return the operator
+   */
+  bool hasBuiltinOperator() const;
+
+  /**
+   * @return true iff this Term has an indexed operator
+   */
+  bool hasOpTermOperator() const;
+
+  /**
+   * @return true iff this Term is the result of
+   *         applying an uninterpreted function
+   */
+  bool hasUFOperator() const;
+
+  /**
+   * @return the OpTerm used to create this time
+   *
+   * valid to call when hasOpTermOperator() returns true
+   */
+  OpTerm getOpTerm() const;
+
+  /**
+   * @return the OpTerm used to create this time
+   *
+   * valid to call when hasUFOperator() returns true
+   */
+  Term getUF() const;
```

Comment:
The name of this function is not really fitting, as above. Maybe try to find a better name?
Also, can you remind me why we even need this?

Reply:
Maybe a solution to the naming problem here and above is to switch the order and use these names:

hasOperatorBuiltin
hasOperatorOpTerm
hasOperatorUF

getOperatorBuiltin
getOperatorOpTerm
getOperatorUF

What do you think?

Reply:
Makes sense to me, good idea! The only issue I have with it is that `getOperatorBuiltin` and `getKind` would do the same thing. Do you think it's more important to have a predictable API (e.g. has `getOperatorBuiltin` to match the other versions), or no redundancy in the API (don't include it because `getKind` already does that). 

I guess we could also just remove `hasOperatorBuiltin` and then you'd have to check that both `hasOperatorOpTerm` and `hasOperatorUF` return false to know that it's a builtin operator. What do you think?

Reply:
> Makes sense to me, good idea! The only issue I have with it is that `getOperatorBuiltin` and `getKind` would do the same thing. Do you think it's more important to have a predictable API (e.g. has `getOperatorBuiltin` to match the other versions), or no redundancy in the API (don't include it because `getKind` already does that).

Hm, yeah, actually we don't want redundancy if possible. But at the same time we want predictability.  I'm a bit torn, tbh, but leaning towards less redundancy. @mpreiner what do you think?

> I guess we could also just remove `hasOperatorBuiltin` and then you'd have to check that both `hasOperatorOpTerm` and `hasOperatorUF` return false to know that it's a builtin operator. What do you think?

No, I'd keep it.